pkg/util/cert/cert.go

```diff
@@ -126,56 +134,84 @@ func MakeEllipticPrivateKeyPEM() ([]byte, error) {
 	return pem.EncodeToMemory(privateKeyPemBlock), nil
 }
 
-// GenerateSelfSignedCertKey creates a self-signed certificate and key for the given host.
-// Host may be an IP or a DNS name
-// You may also specify additional subject alt names (either ip or dns names) for the certificate
-func GenerateSelfSignedCertKey(host string, alternateIPs []net.IP, alternateDNS []string) ([]byte, []byte, error) {
-	priv, err := rsa.GenerateKey(cryptorand.Reader, 2048)
+// GenerateSelfSignedCertKey creates a self-signed CA, a server key and certificate signed
+// with the CA. Host may be an IP or a DNS name. You may also specify additional subject
+// alt names (either ip or dns names) for the certificate.
+func GenerateSelfSignedCertKey(host string, alternateIPs []net.IP, alternateDNS []string) (caCertPem []byte, caKeyPem []byte, certPem []byte, keyPem []byte, err error) {
+	// create root CA
+	glog.Infof("Creating root ca")
+	rootCAReq := csr.CertificateRequest{
+		CN: fmt.Sprintf("%s@ca-%d", host, time.Now().Unix()),
+		KeyRequest: &csr.BasicKeyRequest{
+			A: "rsa",
+			S: 2048,
+		},
+		CA: &csr.CAConfig{
+			Expiry: fmt.Sprintf("%dh", 24*365*1),
+		},
+	}
+	caCertPem, _, caKeyPem, err = initca.New(&rootCAReq)
 	if err != nil {
-		return nil, nil, err
+		return nil, nil, nil, nil, fmt.Errorf("error creating root-ca: %v", err)
```

Comment:
Am guessing end users will see this error, let's spell it out root-ca -> "root certificate authority" if so.